gapic-generator-java/src/main/java/com/google/api/generator/gapic/composer/defaultvalue/DefaultValueComposer.java

```diff
@@ -442,6 +454,79 @@ public static Expr createSimpleMessageBuilderValue(
+   * there are references to createResourceHelperValue being used elsewhere (client header samples,
+   * sample code gen) which is a much larger effort to change.
+   */
+  static ResourceName getMatchingResource(
+      Field field, Map<String, ResourceName> resourceNames, HttpBindings bindings) {
+    if (!field.hasResourceReference()) {
+      return null;
+    }
+    String resourceTypeString = field.resourceReference().resourceTypeString();
+    if (!resourceNames.containsKey(resourceTypeString)) {
+      return null;
+    }
+    List<ResourceName> matchedResources = ImmutableList.of(resourceNames.get(resourceTypeString));
+    // Use the parent resource if the field is a `child_type`. This is for certain RPCs like `List`
+    // and `Create` which have a resource reference to the parent collection.
+    if (field.resourceReference().isChildType()) {
+      // A resource may have multiple parents. Check the possibility for each parent resource
+      List<ResourceName> parentResources =
+          findParentResources(matchedResources.get(0), new ArrayList<>(resourceNames.values()));
+      if (!parentResources.isEmpty()) {
+        matchedResources = parentResources;
+      }
+    }
+
+    // Rationale for the logic below:
+    // If the field is used in the path, check that the field's resource reference matches the RPC's
+    // HttpBindings. There is an edge case where an RPC's HttpBindings defined in the proto may not
+    // match a message. This is possible for Mixin RPCs if service teams configures a custom
+    // HttpBindings in the service yaml file. For example, the GetIamPolicyRequest's resource field
+    // has a resource reference to match anything (`*`). This normally works because the
+    // GetIamPolicy RPC will take in any resource (HttpBindings is defined with `{resource=**}`).
+    // But if the custom path is expected to take in a resource with a more specific path (i.e.
+    // `{resource=service/*}), it won't work unless the resource has a `service/*` pattern. This
+    // logic will try to match a resource that has the pattern `service/*`.
+
+    // Resource reference (google.api.resource_reference) is a string type with a field name of
+    // `parent` or `name`. It won't have anything nested in the bindings (i.e. `{name.field=*}`).
+    // This check is needed as a message can contain fields with valid resource references that
+    // aren't used in the path. If the field isn't used in the path, simply return the matched
+    // resource.
+    boolean isFieldUsedInPath =
+        bindings != null && bindings.getPathParametersValuePatterns().containsKey(field.name());
+    if (isFieldUsedInPath) {
+      for (ResourceName resourceName : matchedResources) {
+        // If the resource is matched as a wildcard `*`, the field's resource_reference is a
+        // wildcard. Attempt to match with a known resource.
+        if (resourceName.isOnlyWildcard()) {
```

Comment:
What exactly does `isOnlyWildcard` mean?

Reply:
I believe If the resource's pattern is `*`

Reply:
Thanks! Do you have an example of it?

Reply:
Asset resource is defined as wildcard pattern: https://github.com/googleapis/googleapis/blob/573205a81b9b96ea6d4e0365373839b59f4b427e/google/cloud/asset/v1/assets.proto#L96-L99

so it could be something with this kind of resource reference: https://github.com/googleapis/googleapis/blob/573205a81b9b96ea6d4e0365373839b59f4b427e/google/cloud/asset/v1/asset_service.proto#L451-L456

The resource_reference for a field can be wildcard: https://github.com/googleapis/googleapis/blob/573205a81b9b96ea6d4e0365373839b59f4b427e/google/iam/v1/iam_policy.proto#L104-L106